### ISSUE TITLE
ModelTrainer class should support resuming training from a checkpoint and allow custom optimizer and loss function

### ISSUE THREAD

**Issue by VecherVhatuX** [VecherVhatuX/learning_data_science_in_prison#1920]
The ModelTrainer class does not handle the case where the model is already trained and the user wants to resume training from a checkpoint, and it also does not allow the user to specify a different optimizer or loss function.
